src/compiler/checker.ts

```diff
@@ -36090,7 +36090,11 @@ export function createTypeChecker(host: TypeCheckerHost): TypeChecker {
     }
 
     function isGenericFunctionReturningFunction(signature: Signature) {
-        return !!(signature.typeParameters && isFunctionType(getReturnTypeOfSignature(signature)));
+        if (!signature.typeParameters) {
+            return false;
+        }
+        const returnType = getReturnTypeOfSignature(signature);
+        return returnType.flags & TypeFlags.Intersection ? some((returnType as IntersectionType).types, isFunctionType) : isFunctionType(returnType);

```

Comment:
Not just `someType`?

Reply:
(or whatever that helper is called that walks intersections and maybe unions)

Reply:
`someType` handles "single" types and unions, not intersections. There is `everyContainedType` that works for unions and intersections but I'd need `someContainedType` here. It can be done here, I just didn't bother.

Reply:
Maybe not in this PR, I guess, but `isUnitLikeType`, `isTypeDerivedFrom`, `isGenericTypeWithUnionConstraint`, `isGenericTypeWithoutNullableConstraint` all do this same thing. (`isEmptyObjectType` copies/pastes the `every` variant` too.)

Reply:
Just pushed out a change introducing `someContainedType`